src/inmanta/plugins.py

```diff
@@ -262,20 +262,19 @@ def to_dsl_type(python_type: type[object]) -> inmanta_type.Type:
     # Unions and optionals
     if typing_inspect.is_union_type(python_type):
         # Optional type
+        bases: Sequence[inmanta_type.Type]
         if typing_inspect.is_optional_type(python_type):
             other_types = [tt for tt in typing.get_args(python_type) if not typing_inspect.is_optional_type(tt)]
```

Comment:
I believe much of this, and especially this list comprehension, won't work with `Annotated.
```python
>>> t = typing.Annotated[int | None, "my-annotation"]
>>> typing_inspect.is_optional_type(t)
False
>>> typing_inspect.is_union_type(t)
False
```

I don't think it has to be a concern for you, but it's something to take into account for @jptrindade's PR.